plugins/nanospell/plugin.js

```diff
@@ -159,6 +164,12 @@
 		init: function (editor) {
 			var self = this;
 
+			// a lock to prevent multiple spellchecks
+			this._isBusy = false;
+
+			// store the current timer
+			this._timer = null;
```

Comment:
Ew. Yup. That would get progressively worse and worse on long documents. Side-note, this would be an excellent usage of a [web worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) to asynchronously spellcheck without causing stuttering on the main UI javascript thread.